app/models/deploy.rb

```diff
@@ -118,6 +118,10 @@ def self.expired
     joins(:job).where(jobs: { status: 'pending'} ).where("jobs.created_at < ?", threshold)
   end
 
+  def url
+    Rails.application.routes.url_helpers.project_deploy_path(project, self)
```

Comment:
Definitely consider presenters/serializers if you need to do this for more things